From c97b64bbc2dc4843bbdcd430ac582634c6b52843 Mon Sep 17 00:00:00 2001
From: Suravee Suthikulpanit <Suravee.Suthikulpanit@amd.com>
Date: Thu, 12 Jun 2014 12:40:24 -0500
Subject: [PATCH 17/38] ahci/xgene: Remove logic to set 64-bit DMA mask
Git-commit: be64beb40df808385f7883d2e6f01ff7c79eeae4
Patch-mainline: v3.17-rc1
References: bnc#902632

Instead of doing the check here, this should be handled
in the common AHCI platform code.

Signed-off-by: Suravee Suthikulpanit <Suravee.Suthikulpanit@amd.com>
Suggested-by: Bartlomiej Zolnierkiewicz <b.zolnierkie@samsung.com>
Cc: Loc Ho <lho@apm.com>
Cc: Tuan Phan <tphan@apm.com>
Cc: Suman Triphati <stripathi@apm.com>
Signed-off-by: Tejun Heo <tj@kernel.org>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 drivers/ata/ahci_xgene.c       | 10 ----------
 drivers/ata/libahci_platform.c |  5 ++---
 2 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/drivers/ata/ahci_xgene.c b/drivers/ata/ahci_xgene.c
index db2fa11..d758735 100644
--- a/drivers/ata/ahci_xgene.c
+++ b/drivers/ata/ahci_xgene.c
@@ -487,16 +487,6 @@ static int xgene_ahci_probe(struct platform_device *pdev)
	/* Configure the host controller */
	xgene_ahci_hw_init(hpriv);

-	/*
-	 * Setup DMA mask. This is preliminary until the DMA range is sorted
-	 * out.
-	 */
-	rc = dma_set_mask_and_coherent(dev, DMA_BIT_MASK(64));
-	if (rc) {
-		dev_err(dev, "Unable to set dma mask\n");
-		goto disable_resources;
-	}
-
 skip_clk_phy:
	hflags = AHCI_HFLAG_NO_PMP | AHCI_HFLAG_NO_NCQ;

--
2.1.0
